highfidelity/hifi
# Fix caching of ktx texture files

https://highfidelity.manuscript.com/f/cases/10533/Major-GPU-burps-when-entering-avatar-island-maybe-texture-related

Our KtxStorage file caching was broken.  Files would always be closed immediately after use, leading to poor performance times during texture creation (red line is 12 seconds):
![cachefix1](https://user-images.githubusercontent.com/7650116/34066315-d6b8c7bc-e1c1-11e7-8695-f65386233398.png)
Single textures repeatedly being loaded from scratch:
![cachefix2](https://user-images.githubusercontent.com/7650116/34066336-1f9a2dcc-e1c2-11e7-8314-0eb07a989bb2.png)
Now, we keep the file open for one frame and then close it.  This reduces impact on the present thread:
![cachefix3](https://user-images.githubusercontent.com/7650116/34066408-eb727242-e1c2-11e7-9f6a-b96abefa3ff4.png)
There are still small performance hits that will be investigated separately.

To test:
- With master, clear your caches (delete AppData/Local/High Fidelity).  Open Interface in your sandbox.  Go to AvatarIsland (if there's a protocol mismatch, you can try leviathan or SamsWorld if they are up, otherwise I can send you the models file).
- Your game rate should drop drastically.  Without moving, note how long it takes to stabilize (if your machine is strong enough, should stabilize at 60).  It takes 12 second on my computer, but might be different for you, possibly depending on what type of hard drive you have (?).
- Repeat the above steps with this PR (make sure to clear your cache).  The time should be much lower.  All content and textures should still load if you wait long enough.  In general, the loading experience should not impact game rate as much.
